java/connector-node/risingwave-sink-jdbc/src/main/java/com/risingwave/connector/JDBCSinkFactory.java

```diff
@@ -47,12 +53,56 @@ public void validate(TableSchema tableSchema, Map<String, String> tablePropertie
         }
 
         String jdbcUrl = tableProperties.get(JDBC_URL_PROP);
+        String tableName = tableProperties.get(TABLE_NAME_PROP);
+        Set<String> jdbcColumns = new HashSet<>();
+        Set<String> jdbcPk = new HashSet<>();
 
         try {
             Connection conn = DriverManager.getConnection(jdbcUrl);
+
+            ResultSet columnResultSet = conn.getMetaData().getColumns(null, null, tableName, null);
+            while (columnResultSet.next()) {
+                jdbcColumns.add(columnResultSet.getString("COLUMN_NAME"));
+            }
+
+            ResultSet pkResultSet = conn.getMetaData().getPrimaryKeys(null, null, tableName);
+            while (pkResultSet.next()) {
+                jdbcPk.add(pkResultSet.getString("COLUMN_NAME"));
+            }
+
             conn.close();
         } catch (SQLException e) {
             throw Status.INTERNAL.withCause(e).asRuntimeException();
         }
```

Comment:
Let's use `try-with-resources` to improve the code here.
https://stackoverflow.com/questions/8066501/how-should-i-use-try-with-resources-with-jdbc

Reply:
May do the same for iceberg Hadoop catalog of possible?